packages/quantic/force-app/main/default/lwc/quanticRadioButtonsGroup/__tests__/quanticRadioButtonsGroup.test.js

```diff
@@ -0,0 +1,289 @@
+      // @ts-ignore
+      const element = createTestComponent({
+        ...exampleOptions,
+        hideLabels: false,
+      });
+      await flushPromises();
+
+      const radioLabels = element.shadowRoot.querySelectorAll(
+        selectors.radioLabels
+      );
+
+      expect(radioLabels).toHaveLength(defaultOptionsList.length);
+      for (const radioLabel of radioLabels) {
+        expect(radioLabel.classList.contains(labelHiddenClass)).toBeFalsy();
+      }
+    });
+  });
+
+  describe('User interaction', () => {
+    it('should dispatch the "change" event when a radio button is clicked', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const radioInputs = element.shadowRoot.querySelectorAll(
+        selectors.radioInputs
+      );
+      const radioInput = radioInputs[0];
+      const changeHandler = jest.fn();
+      element.addEventListener('change', changeHandler);
+
+      radioInput.click();
+
+      expect(changeHandler).toHaveBeenCalledTimes(1);
+      expect(changeHandler.mock.calls[0][0].detail).toEqual({
+        value: radioInput.value,
+      });
+    });
+
+    it('should update the selected radio button when a radio button is clicked', async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const radioInputs = element.shadowRoot.querySelectorAll(
+        selectors.radioInputs
+      );
+      const radioButtonContainers = element.shadowRoot.querySelectorAll(
+        selectors.radioContainers
+      );
+      const radioInput = radioInputs[0];
+      const radioContainer = radioButtonContainers[0];
+
+      radioInput.click();
+      await flushPromises();
+
+      expect(radioInput.checked).toBeTruthy();
+      expect(
+        radioContainer.classList.contains('radio-button--selected')
+      ).toBeTruthy();
+    });
+
+    it("shouldn't be possible to select multiple radio buttons", async () => {
+      const element = createTestComponent();
+      await flushPromises();
+
+      const radioInputs = element.shadowRoot.querySelectorAll(
+        selectors.radioInputs
+      );
+      const radioButtonContainers = element.shadowRoot.querySelectorAll(
+        selectors.radioContainers
+      );
+
+      radioInputs[0].click();
+      radioInputs[1].click();
+      await flushPromises();
+
+      const selectedRadioInputs = element.shadowRoot.querySelectorAll(
+        `.radio-button--selected`
+      );
+
+      expect(radioInputs[0].checked).toBeFalsy();
+      expect(radioInputs[1].checked).toBeTruthy();
+      expect(
+        radioButtonContainers[0].classList.contains('radio-button--selected')
+      ).toBeFalsy();
+      expect(
+        radioButtonContainers[1].classList.contains('radio-button--selected')
+      ).toBeTruthy();
+      expect(selectedRadioInputs).toHaveLength(1);
+    });
+
+    it('should dispatch the "focus" event when a radio button is focused', async () => {
```

Comment:
Is it possible to have the same test, but this time for the blur event?

Reply:
I don't really know how to trigger a blur event on a radio input, tbh I'm not even sure we _can_ blur on a radio button.
I only put it there because I took the same mechanics as the original LWC radio-buttons-group to start with and it was there.